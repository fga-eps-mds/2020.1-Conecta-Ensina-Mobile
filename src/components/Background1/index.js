import React, {useContext} from 'react';
import Theme from '../../../Theme';
import CustomText from '../CustomText';
import {
  BlueContainer,
  WhiteContainer,
  TextoContainer,
  UserAvatar,
  UserImage,
  UserContainer,
} from './styles';
import {AuthContext} from '../../contexts/auth';

export default function Background1({children}) {
  const {user} = useContext(AuthContext);

  return (
    <Theme>
      <BlueContainer>
        <UserContainer>
          <UserAvatar>
            <UserImage source={require('../../assets/user_blue.png')} />
          </UserAvatar>
          <TextoContainer>
            <CustomText white medium>
              {user && user.firstName}
            </CustomText>
          </TextoContainer>
        </UserContainer>
        <WhiteContainer>{children}</WhiteContainer>
      </BlueContainer>
    </Theme>
  );
}
