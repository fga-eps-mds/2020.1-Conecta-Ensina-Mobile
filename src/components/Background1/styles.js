import styled from 'styled-components/native';

export const BlueContainer = styled.View`
  background-color: ${(props) => props.theme.colors.fundoAzul};
  flex: 1;
  display: flex;
  justify-content: flex-end;
`;

export const WhiteContainer = styled.View`
  background-color: ${(props) => props.theme.colors.branco};
  flex: 78;
  display: flex;
  justify-content: center;
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
`;

export const UserContainer = styled.TouchableOpacity`
  flex-direction: row-reverse;
  flex: 22;
  margin-left: 30px;
  align-items: center;
`;

export const TextoContainer = styled.View`
  margin-bottom: 40px;
  margin-right: 35px;
`;

export const UserAvatar = styled.TouchableOpacity`
  width: 58px;
  height: 58px;
  border-radius: 1000px;
  background-color: ${(props) => props.theme.colors.branco};
  margin-left: 40px;
  margin-bottom: 30px;
  justify-content: center;
  align-items: center;
`;

export const UserImage = styled.Image`
  width: 33px;
  height: 38px;
`;

export const ContainerUpText = styled.TouchableOpacity`
  flex-direction: row-reverse;
  flex: 10;
  padding: 10%;
  margin-bottom: 5%;
  align-items: center;
`;
