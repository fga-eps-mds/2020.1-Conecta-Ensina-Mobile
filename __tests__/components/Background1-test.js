import React from 'react';
import Background1 from '../../src/components/Background1';
import {render, fireEvent} from '@testing-library/react-native';

describe('Testing Background1', () => {
  const navigation = {
    navigate: jest.fn(),
  };

  test('Should userAvatar be pressed', () => {
    const {getByTestId} = render(<Background1 navigation={navigation} />);

    const userAvatar = getByTestId('userAvatar');

    fireEvent.press(userAvatar);
    expect(navigation.navigate).toBeCalled();
  });

  test('Should match snapshot', () => {
    const tree = render(<Background1 />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
