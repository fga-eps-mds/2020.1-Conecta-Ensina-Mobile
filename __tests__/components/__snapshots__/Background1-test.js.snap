// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#184E79",
        "display": "flex",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "row-reverse",
        "flexGrow": 22,
        "flexShrink": 1,
        "marginLeft": 30,
        "opacity": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 1000,
          "height": 58,
          "justifyContent": "center",
          "marginBottom": 30,
          "marginLeft": 40,
          "opacity": 1,
          "width": 58,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/user_blue.png",
          }
        }
        style={
          Array [
            Object {
              "height": 38,
              "width": 33,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "marginBottom": 40,
            "marginRight": 35,
          },
        ]
      }
    >
      <Text
        medium={true}
        style={
          Array [
            Object {
              "color": "#ffffff",
              "fontFamily": "Roboto",
              "fontSize": 25,
              "textAlign": "center",
            },
          ]
        }
        white={true}
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "display": "flex",
          "flexBasis": 0,
          "flexGrow": 78,
          "flexShrink": 1,
          "justifyContent": "center",
        },
      ]
    }
  />
</View>
`;
